app/src/main/kotlin/org/imperial/mrc/hint/AppProperties.kt

```diff
@@ -0,0 +1,42 @@
+package org.imperial.mrc.hint
+
+import org.springframework.context.annotation.Configuration
+import java.io.File
+import java.io.FileNotFoundException
+import java.net.URL
+import java.util.*
+
+interface AppProperties {
+    val uploadDirectory: String
+}
+
+@Configuration
+class ConfiguredAppProperties: AppProperties {
+
+    override val uploadDirectory = properties["upload_dir"].toString()
+
+    companion object
+    {
+        val properties = Properties().apply {
+            load(getResource("config.properties").openStream())
+            val global = File("/etc/orderly/web/config.properties")
```

Comment:
Should be /etc/hint..
Would this class be more testable if the path orproperties could be injected?

Reply:
Not sure what you mean by this class being testable?

Reply:
Ah you mean to catch typos of this kind? Sure